File: exampleApp/src/main/java/it/trade/android/exampleapp/PositionsActivity.java
package it.trade.android.exampleapp;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.method.ScrollingMovementMethod;
import android.widget.TextView;

import java.util.List;

import it.trade.android.sdk.model.TradeItLinkedBrokerAccountParcelable;
import it.trade.android.sdk.model.TradeItPositionParcelable;
import it.trade.model.TradeItErrorResult;
import it.trade.model.callback.TradeItCallback;

import static it.trade.android.exampleapp.MainActivity.POSITIONS_PARAMETER;


public class PositionsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_positions);
        final TextView textView = (TextView) this.findViewById(R.id.positions_textview);
        textView.setMovementMethod(new ScrollingMovementMethod());
        Intent intent = getIntent();
        List<TradeItPositionParcelable> positions = intent.getParcelableArrayListExtra(POSITIONS_PARAMETER);
        textView.setText(positions.toString());
        TradeItLinkedBrokerAccountParcelable linkedBrokerAccount = intent.getParcelableExtra(MainActivity.PARCELED_ACCOUNT_PARAMETER);
        TradeItPositionParcelable position = null;
        for (TradeItPositionParcelable positionParcelable: positions) {
            if (positionParcelable.isProxyVoteEligible) {
                position = positionParcelable;
                break;
            }
        }
        linkedBrokerAccount.getProxyVoteUrl(position.symbol, new TradeItCallback<String>() {
            @Override
            public void onSuccess(String proxyVoteUrl) {
                textView.setText("Proxyvote url for last position: " + proxyVoteUrl);
            }

            @Override
            public void onError(TradeItErrorResult error) {
                textView.setText("Error getting Proxyvote url for last position: " + error);
            }
        });
    }
}

File: exampleApp/src/main/java/it/trade/android/exampleapp/PositionsActivity.kt
package it.trade.android.exampleapp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.method.ScrollingMovementMethod
import android.widget.TextView
import it.trade.android.sdk.model.TradeItPositionParcelable


class PositionsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_positions)
        val textView = this.findViewById(R.id.positions_textview) as TextView?
        textView!!.movementMethod = ScrollingMovementMethod()
        val intent = intent
        val positions = intent.getParcelableArrayListExtra<TradeItPositionParcelable>(MainActivity.POSITIONS_PARAMETER)
        textView.text = positions.toString()
    }
}
